src/index.js

```diff
@@ -29,97 +27,137 @@ function handleConsentResult(tcfApi, store = {}, vendorList, consentData = {}) {
 		policyVersion: consentPolicyVersion = 1
 	} = consentData;
 
-	const tcfApiCall = (command) => {
-		tcfApi(command, 2, () => log.debug(`${command} command was called`));
-	};
+	let displayOptions = {};
+
+	if (!config.autoDisplay) {
+		return { display: false };
+	}
 
 	if (!created) {
 		log.debug('No consent data found. Showing consent tool');
-		config.autoDisplay && tcfApiCall('showConsentTool');
+		displayOptions = {display: true, fn: 'showConsentTool'};
 	} else if (!listVersion) {
 		log.debug('Could not determine vendor list version. Not showing consent tool');
+		displayOptions = { display: false };
 	} else if (vendorListVersion !== listVersion) {
 		log.debug(`Consent found for version ${vendorListVersion}, but received vendor list version ${listVersion}. Showing consent tool`);
-		config.autoDisplay && tcfApiCall('showConsentTool');
+		displayOptions = {display: true, fn: 'showConsentTool'};
 	} else if (consentPolicyVersion !== listPolicyVersion) {
 		log.debug(`Consent found for policy ${consentPolicyVersion}, but received vendor list with policy ${consentPolicyVersion}. Showing consent tool`);
-		config.autoDisplay && tcfApiCall('showConsentTool');
+		displayOptions = {display: true, fn: 'showConsentTool'};
 	} else {
 		log.debug('Consent found. Not showing consent tool. Show footer when not all consents set to true');
-		!isConsentToolShowing && config.autoDisplay && tcfApiCall('showFooter');
+		displayOptions = {display: false, fn: 'showFooter'};
 	}
-}
 
-function checkConsent(tcfApi, store) {
-	if (!tcfApi) {
-		log.error('CMP failed to load');
-	}
-	else if (!window.navigator.cookieEnabled) {
-		log.warn('Cookies are disabled. Ignoring CMP consent check');
-	}
-	else {
-		const resolve = (timeout, consentData = {}, vendorList) => {
-			clearTimeout(timeout);
-			handleConsentResult(tcfApi, store, vendorList, consentData);
-		};
-
-		const { getVendorList, getConsentData } = config;
-		if (getVendorList) {
-			getVendorList((err, vendorList) => {
-				if (err) {
-					log.error('Failed to get vendor list');
-				} else {
-					const timeout = setTimeout(() => {
-						handleConsentResult(tcfApi, store, vendorList);
-					}, 100);
+	return  displayOptions;
+};
 
-					if (getConsentData) {
-						getConsentData((err, data) => {
-							if (err) {
-								resolve(timeout);
-							} else {
-								try {
-									const tcStringDecoded = decodeConsentData(data.consent);
-									resolve(timeout, tcStringDecoded, vendorList);
-								} catch (e) {
+
+const shouldDisplay = (tcfApi) => {
+	return new Promise((resolve, reject) => {
+		if (!tcfApi) {
+			const msg = 'CMP failed to load';
+			log.error(msg);
+			reject(msg);
```

Comment:
Do we really have to reject here? This code in fact validates whether `__tcfapi` stub has been created. But before this pull request we didn't do it. We validated only whether full `__tcfapi` implementation has been initialized. And now we should also do it but inside `displayUI` function.